### PR TITLE
Uncomment types for Record, fix for .equals

### PR DESCRIPTION
**Pull request by nmn**
The Record type refs work as of Flow version 0.24.2.

The .equals method on `_Iterable` was causing Set.equals to not accept another Set. This seems to fix it.

### CODE REVIEW

File: type-definitions/immutable.js.flow [hunk 1]
```diff
@@ -619,13 +619,11 @@ declare class Stack<T> extends IndexedCollection<T> {
 declare function Range(start?: number, end?: number, step?: number): IndexedSeq<number>;
 declare function Repeat<T>(value: T, times?: number): IndexedSeq<T>;
 
-//TODO: Once flow can extend normal Objects we can change this back to actually reflect Record behavior.
-// For now fallback to any to not break existing Code
 declare class Record<T: Object> {
-  static <T: Object>(spec: T, name?: string): /*T & Record<T>*/any;
+  static <T: Object>(spec: T, name?: string): T & Record<T>;
```
Comment (tallixun) [inline type-definitions/immutable.js.flow]: Hi, are you sure that this line is correct? Isn't it supposed to be (I have checked it only with flow 0.26, so maybe it is working with 0.24):

```
static <T: Object>(spec: T, name?: string): Class<T & Record<T>>;
```

Comment (nmn) [inline type-definitions/immutable.js.flow]: Yes, you're right about this. I checked the other cases, and so I just uncommented the things I saw. I'll fix this.

**Comment by nmn** [comment]
Hold off on merging for now. Thanks @tallixun

**Comment by nmn** [comment]
Fixed outstanding issues. Added the `getIn` and `setIn` methods for Records.
If you have any suggestions for improving the type definitions let me know.

@tallixun would love another review

**Comment by philipp-spiess** [comment]
Hey @nmn. I would love to see your PR merged. However, there are a few method definitions missing. So far I found:

- `Record#updateIn()`
- `Record#toJS()`
- `Record#toObject()`
- `Record#update()
- `Record#withMutations()`
- `Record#delete()`
- `Record#merge()`
 
I think it would make sense to extend/mix in `KeyedCollection`, since this is also the way it is implemented: https://github.com/facebook/immutable-js/blob/master/src/Record.js#L18

**Comment by lacker** [comment]
So https://github.com/facebook/immutable-js/pull/878 refactored the flow types and also added tests for them. Would you mind updating this PR to merge those changes in, and also add a test that catches this? Thanks!

**Comment by kozlitinaelja** [comment]
Hi there. I am starting handling backlog. I know it is been a while since you've submitted PR, sorry for such delay. The are some conflicts. Could you, please, rebase? 😅

**Pull request closed without merging**